### PR TITLE
chore(codegen): use parallel bazel target building

### PR DESCRIPTION
**Pull request by diegomarquezp**
So far, there was a `git reset --hard` performed between each call to `generate-one.sh`. The following changes were made to take advantage of bazel's parallel building capabilities.
* Leave `generate-one.sh` as a post processing script (after bazel targets are build)
* Move the preparation logic of `generate-one.sh` to `setup-build-rule.sh`, which adds the necessary rules for building
* `generate-all.sh` now loops twice over the library list; one time with `setup-build-rule.sh` and a second time with `generate-one.sh`.
* Between these two calls, a single call to `bazel build` is made against all `.*java_gapic_spring` targets

### CODE REVIEW

**Comment by meltsufin** [comment]
@diegomarquezp Is this ready for review?

**Review by emmileaf — APPROVED** [review]
Thank you @diegomarquezp, this is a great improvement!

One question I have that doesn’t need to block this PR: 
- Since the bazel builds are now moved to `setup-build-rules.sh`, is it correct to understand that any failures will exit this script immediately, instead of being captured by the error handling in #1437? If so, is there a way to have the existing error handling set up work with the parallel bazel building, or would it be a more complex change?

File: spring-cloud-generator/generate-all.sh [hunk 1]
```diff
@@ -20,7 +21,23 @@ save_error_info () {
...
 
 bash download-repos.sh
+bash setup-googleapis-rules.sh
 libraries=$(cat $WORKING_DIR/library_list.txt | tail -n+2)
+
+
+# modifies the BUILD files of each of the entries in the library list
+while IFS=, read -r library_name googleapis_location coordinates_version googleapis_commitish monorepo_folder; do
+  echo "preparing bazel rules for $library_name"
+  bash $WORKING_DIR/setup-build-rule.sh \
```
[the remark this answers is not in the document]
Comment (diegomarquezp) [reply spring-cloud-generator/generate-all.sh:31]: @emmileaf The bazel build has indeed been moved outside `generate-one.sh`, but it is now living between the two for loops of this script (`generate-all.sh:38`). If I understood your point correctly, I think the current name for the build rule script should be OK.

**Comment by diegomarquezp** [comment]
> Thank you @diegomarquezp, this is a great improvement!
> 
> One question I have that doesn’t need to block this PR:
> 
> * Since the bazel builds are now moved to `setup-build-rules.sh`, is it correct to understand that any failures will exit this script immediately, instead of being captured by the error handling in #1437? If so, is there a way to have the existing error handling set up work with the parallel bazel building, or would it be a more complex change?

Thanks for pointing this out. Yes, calls to `setup-build-rules.sh` may break the whole `generate-all.sh` due to `set -e` on both scripts. Luckily, we can catch a non-zero exit both in the parallel bazel call and in `setup-build-rules.sh` loop call by adding `[command] || save_error_info [error_title]`. I think we can leave a more sophisticated implementation of error handling as a separate task, but I think we need to add this. I will quickly push a fix in this PR

**Pull request merged**